go/cmd/zk/zkcmd.go

```diff
@@ -193,12 +196,14 @@ func isZkFile(path string) bool {
 	return strings.HasPrefix(path, "/zk")
 }
 
-func cmdWait(ctx context.Context, subFlags *flag.FlagSet, args []string) error {
+func cmdWait(ctx context.Context, subFlags *pflag.FlagSet, args []string) error {
 	var (
 		exitIfExists = subFlags.Bool("e", false, "exit if the path already exists")
 	)
 
-	subFlags.Parse(args)
+	if err := subFlags.Parse(args); err != nil {
+		return err
+	}
```

Comment:
because you specified `ExitOnError` for the sub flags, this will never return with a non-nil error (it will, well, exit, instead)

Reply:
(to be clear, i think you want to keep this bit, but change the sub flags to `ContinueOnError`)

Reply:
done